src/commands/channels/publish.ts

```diff
@@ -87,8 +87,11 @@ export default class ChannelsPublish extends AblyBaseCommand {
   async run(): Promise<void> {
     const { args, flags } = await this.parse(ChannelsPublish);
 
-    // Use REST by default now - only create Realtime client if explicitly requested
-    await (flags.transport === "realtime"
+    // Use Realtime transport by default when publishing multiple messages to ensure ordering
+    const shouldUseRealtime = flags.transport === "realtime" || 
+      (flags.transport !== "rest" && flags.count > 1);
```

Comment:
The default transport is `rest`, so the only time we're going to use "realtime" is if the "--transport realtime" flag is passed, which is what the old check did. I think it was clearer before. 

```
const shouldUseRealtime = flags.transport === "realtime"
```

flags | transport used
-------------|-------------
--transport "realtime" --count 3 | realtime
--transport "rest" --count 3 | rest
--count 3 | rest (default)